tools/score_instrument.php

```diff
@@ -145,41 +164,74 @@ function log_msg($message) {
 
         // check if the instrument has a scoring method
         if (!method_exists($instrument, "score")) {
-            fwrite(STDERR, "Error, the instrument ($test_name) does not have a scoring feature \n");
-            return false;
+            printOut("Error, the instrument ($test_name) does not have a scoring feature");
+            continue;
         }
 
         // print out candidate/session info
-        fwrite(STDERR, "Candidate: ".$record['CandID']."/".$record['Visit_label']."/".$record['SessionID'].":: (".$record['PSCID'].")\n");
-        //fwrite(STDERR, "Candidate: ".$instrument->_dob."/"."Test_name:".$test_name."/". $instrument->_pls3Age."/".$instrument->getDateOfAdministration().":: \n");
+        printOut("Candidate: ".$record['CandID']."/".$record['Visit_label']."/".$record['SessionID'].":: (".$record['PSCID'].")");
 
         // call the score function
-        $oldRecord = $db->pselectRow("SELECT * FROM $test_name WHERE CommentID=:cid", array('cid' => $record['CommentID']));
+        $oldRecord = $DB->pselectRow(
+            "SELECT * FROM $test_name WHERE CommentID=:cid",
+            array('cid' => $record['CommentID'])
+        );
         $success = $instrument->score();
-        $newRecord = $db->pselectRow("SELECT * FROM $test_name WHERE CommentID=:cid", array('cid' => $record['CommentID']));
+        $newRecord = $DB->pselectRow(
+            "SELECT * FROM $test_name WHERE CommentID=:cid",
+            array('cid' => $record['CommentID'])
+        );
         unset($oldRecord['Testdate']);
         unset($newRecord['Testdate']);
         $diff = array_diff_assoc($oldRecord, $newRecord);
 
         if($diff != array()) {
 
-            log_msg("Updated $record[CommentID]:");
+            logMessage("Updated $record[CommentID]:");
             foreach ($diff as $key => $val) {
                 $old = $oldRecord[$key] == null ? 'null' : $oldRecord[$key];
                 $new = $newRecord[$key] == null ? 'null' : $newRecord[$key];
-                log_msg("\t$key: $old => $new");
+                logMessage("\t$key: $old => $new");
             }
         }
         else {
-            log_msg("No changes made to " . $record['CommentID']);
+            logMessage("No changes made to " . $record['CommentID']);
         }
 
-        fwrite(STDERR, "-- OK! \n");
+        printOut("-- OK!");
         // unset
         unset($instrument);
     }
 }
-
 fclose($logfp);
-fwrite(STDERR, "End \n")
-?>
+printOut("End");
+
+/*
+ * Prints to log file
+ */
+function logMessage($message) {
+    global $logfp;
+    if (!$logfp) {
+        //The log file could not be instanciated
+        //use print instead
+        printOut($message);
+    }
+    $now_array = getdate();
+    $now_string = sprintf("%04d-%02d-%02d %02d:%02d:%02d", $now_array['year'], $now_array['mon'], $now_array['mday'], $now_array['hours'], $now_array['minutes'], $now_array['seconds']);
+    fwrite($logfp, "[$now_string] $message\n");
```

Comment:
could use strftime(). Would be easier to read and can skip the $now_array

$now = strftime("%Y-%m-%d %H:%M:%S")

Reply:
talk to me about risk of unexpected behaviour, is sprintf worst than strftime ?

Reply:
Not sure about unexpected behaviour for each function but I can say that using strftime() is more specialised and have less case that sprintf(). Combine that the fact that you don't have to call getdate(), create an array for one time use,...

easy guest is that strftime() is not only faster but have less risk of unexpected behaviour.

Reply:
Should be easy to implement quickly (same for line 34-35). Understand that this PR is for a quick fix but it is not for extreme urgency.

Reply:
it is for CCNA since we need to restore all instruments at once. but is okai we made an override incorporating this fix

@davidblader this PR can be closed now. Since you issued it I'll leave it to you.